phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/GroupedAggregateRegionObserver.java

```diff
@@ -230,10 +238,17 @@ private static final class InMemoryGroupByCache implements GroupByCache {
         private final ServerAggregators aggregators;
         private final RegionCoprocessorEnvironment env;
         private final byte[] customAnnotations;
+        private final ConcurrentMap<ImmutableBytesWritable, ImmutableBytesWritable>
```

Comment:
Any reason why you are using `ImmutableBytesWritable` vs aggregateMap is using `ImmutableBytesPtr` ?

Reply:
No reason as such, ImmutableBytesPtr extends ImmutableBytesWritable anyway, and both have similar hashcode and equals functions. We can use either one, `ImmutableBytesWritable` is more generic and while retrieving key/value pair from the map, we actually use `ImmutableBytesWritable` type data:
```
            for (Map.Entry<ImmutableBytesPtr, Aggregator[]> entry : aggregateMap.entrySet()) {
                ImmutableBytesWritable aggregateGroupValPtr = entry.getKey();
                Aggregator[] rowAggregators = entry.getValue();
```

Hence, even though aggregateMap has `ImmutableBytesPtr` as key, we still use `ImmutableBytesWritable` as generic key type.